diff --git b/lwip/src/core/ipv4/dhcp.c a/lwip/src/core/ipv4/dhcp.c
index 76b9f682..13858a63 100644
--- b/lwip/src/core/ipv4/dhcp.c
+++ a/lwip/src/core/ipv4/dhcp.c
@@ -1826,7 +1826,7 @@ dhcp_recv(void *arg, struct udp_pcb *pcb, struct pbuf *p, const ip_addr_t *addr,
     /* in requesting state? */
     if (dhcp->state == DHCP_STATE_REQUESTING) {
       dhcp_handle_ack(netif, msg_in);
-#if DHCP_DOES_ARP_CHECK
+#if 0 //DHCP_DOES_ARP_CHECK
       if ((netif->flags & NETIF_FLAG_ETHARP) != 0) {
         /* check if the acknowledged lease address is already in use */
         dhcp_check(netif);
